platform/storage/posix/src/mender-storage.c

```diff
@@ -315,6 +316,38 @@ mender_storage_delete_provides(void) {
 #endif /*CONFIG_MENDER_FULL_PARSE_ARTIFACT*/
 #endif /*CONFIG_MENDER_PROVIDES_DEPENDS*/
 
+mender_err_t
+mender_storage_set_artifact_name(const char *artifact_name) {
+
+    assert(NULL != artifact_name);
+
+    size_t artifact_name_str_length = strlen(artifact_name);
+
+    if (MENDER_OK != mender_storage_write_file(MENDER_STORAGE_NVS_ARTICACT_NAME, artifact_name, artifact_name_str_length)) {
+        return MENDER_FAIL;
+    }
+    return MENDER_OK;
+}
+
+mender_err_t
+mender_storage_get_artifact_name(char **artifact_name) {
+
+    assert(NULL != artifact_name);
+
+    size_t       artifact_name_length;
+    mender_err_t ret = mender_storage_read_file(MENDER_STORAGE_NVS_ARTICACT_NAME, (void **)artifact_name, &artifact_name_length);
```

Comment:
Looks like this will not read nor add the terminating `\0` byte.

Reply:
The terminating `\0` byte is added in `mender_storage_read_file()`